src/redis-cli.c

```diff
@@ -2087,6 +2089,12 @@ static int parseOptions(int argc, char **argv) {
             config.set_errcode = 1;
         } else if (!strcmp(argv[i],"--verbose")) {
             config.verbose = 1;
+        } else if (!strcmp(argv[i],"-4")) {
+            config.prefer_ipv4 = 1;
+            config.prefer_ipv6 = 0;
+        } else if (!strcmp(argv[i],"-6")) {
+            config.prefer_ipv4 = 0;
+            config.prefer_ipv6 = 1;
```

Comment:
What if both flags are specified? I think we should either use AF_UNSPEC in this case (i.e. prefer both) or return an error for incompatible CLI options.

Reply:
My initial thought was that later option can override the former option (a way of being lazy...)

Reply:
I have seen PRs trying to remove such ambiguities but it's hard to change existing behaviour. It's easier to avoid ambiguities when we're adding new options, like here.

It is very easy to do this: Let -4 only set prefer_ipv4 = 1 and let -6 only set prefer_ipv6 = 1. Then where we call getaddrinfo, just check both flags:

```C
    if (flags & ANET_PREFER_IPV4 && !(flags & ANET_PREFER_IPV6)) {
        hints.ai_family = AF_INET;
    } else if (flags & ANET_PREFER_IPV6 && !(flags & ANET_PREFER_IPV4)) {
        hints.ai_family = AF_INET6;
    }
```

Reply:
i see, this make sense, thanks